diff --git a/include/new b/include/new
index 4e52750..a4b377c 100644
--- a/include/new
+++ b/include/new
@@ -160,6 +160,9 @@ public:
 
 #endif  // defined(_LIBCPP_BUILDING_NEW) || (_LIBCPP_STD_VER > 11)
 
+#if defined(_LIBCPP_ABI_MICROSOFT) && defined(_HAS_ALIGNED_NEW)
+/// vcruntime_new.h defined align_val_t
+#else
 #if !defined(_LIBCPP_HAS_NO_ALIGNED_ALLOCATION) || _LIBCPP_STD_VER > 14
 #ifndef _LIBCPP_CXX03_LANG
 enum class _LIBCPP_ENUM_VIS align_val_t : size_t { };
@@ -167,6 +170,7 @@ enum class _LIBCPP_ENUM_VIS align_val_t : size_t { };
 enum align_val_t { __zero = 0, __max = (size_t)-1 };
 #endif
 #endif
+#endif
 
 }  // std
 
